### ISSUE TITLE
redshift dialect: missing support for ALTER TABLE

### ISSUE THREAD

**Issue by 6e2f61** [sqlfluff/sqlfluff#3391]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Redshift supports `ALTER TABLE` statement according to their [documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_TABLE.html).

sqlfluff fails while parsing files containing this statement.

### Expected Behaviour

Add support for the `ALTER TABLE` statements.

### Observed Behaviour

Not currently supported, results in parse errors.

```
== [scripts/redshift/ddl/schema_name/core_table.sql] FAIL                                                                                                                                   
L:  5 | P:   1 |  PRS | Line 5, Position 1: Found unparsable section: 'alter table example_table\n ...'
All Finished 📜 🎉!
```

### How to reproduce

Create a test file containing as below and try to lint/parse with sqlfluff.

```
CREATE TABLE example_table (
    example_sk VARCHAR(150) NOT NULL
    , example_id VARCHAR(2) NULL );

ALTER TABLE example_table
    ADD CONSTRAINT example_name PRIMARY KEY (example_sk);
```

### Dialect

redshift

### Version

```
sqlfluff, version 0.13.2
Python 3.8.10
```

### Configuration

```
# https://docs.sqlfluff.com/en/stable/configuration.html#config

[sqlfluff]
dialect = redshift
rules = all
exclude_rules = L034
# Controls SQLFluff output, see max_line_length for SQL output
output_line_length = 118
ignore_templated_areas = true
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = true

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 118
indent_unit = space
comma_style = leading

# https://docs.sqlfluff.com/en/stable/rules.html#ruleref

[sqlfluff:rules:L003]
hanging_indents = true

[sqlfluff:rules:L007]
operator_new_lines = after

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper

[sqlfluff:rules:L011]
# of tables
aliasing = explicit

[sqlfluff:rules:L012]
# of columns
aliasing = explicit

[sqlfluff:rules:L013]
allow_scalar = false

[sqlfluff:rules:L019]
comma_style = leading

[sqlfluff:rules:L022]
comma_style = leading

[sqlfluff:rules:L029]
unquoted_identifiers_policy = all
ignore_words = character_maximum_length, column_name, columns, constraint_name, constraint_schema, data_type, table_name

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

[sqlfluff:rules:L031]
# Avoid table aliases in from clauses and join conditions
force_enable = true

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]
capitalisation_policy = upper

[sqlfluff:rules:L051]
fully_qualify_join_types = both

[sqlfluff:rules:L052]
multiline_newline = false
require_final_semicolon = false

[sqlfluff:rules:L054]
group_by_and_order_by_style = explicit

[sqlfluff:rules:L057]
additional_allowed_characters = "{}"
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = false

[sqlfluff:rules:L059]
prefer_quoted_identifiers = false

[sqlfluff:rules:L063]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by menzenski** [comment]
I'm also running into this - specifically, the `CONSTRAINT <constraint name>` seems to be the issue. This SQL is parsed fine:

```sql
CREATE TABLE example_table (
    example_sk VARCHAR(150) NOT NULL
    , example_id VARCHAR(2) NULL );

ALTER TABLE example_table
    ADD PRIMARY KEY (example_sk);
```

**Comment by tunetheweb** [comment]
Fix proposed in #3395
